shell/renderer/api/electron_api_context_bridge.cc

```diff
@@ -383,22 +388,27 @@ v8::MaybeLocal<v8::Value> PassValueToOtherContext(
     return v8::MaybeLocal<v8::Value>(cloned_arr);
   }
 
-  // Custom logic to "clone" Element references
-  blink::WebElement elem =
-      blink::WebElement::FromV8Value(destination_context->GetIsolate(), value);
-  if (!elem.IsNull()) {
-    v8::Context::Scope destination_context_scope(destination_context);
-    return v8::MaybeLocal<v8::Value>(
-        elem.ToV8Value(destination_context->GetIsolate()));
-  }
+  // Clone certain DOM APIs only within Window contexts.
+  blink::ExecutionContext* execution_context =
+      blink::ExecutionContext::From(destination_context);
```

Comment:
I think we want to be checking the `source_context` here?

Reply:
My initial thinking might have been that the source context being a ShadowRealm (in the case of service worker preload realms) is not unique enough since ShadowRealms can be initialized in any context. However, thinking about this again, your suggestion seems more correct.

If we supported DOM context -> non-DOM context value passing, we should be checking the source context 👍